package/gluon-core/luasrc/lib/gluon/upgrade/115-swconfig

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/lua
+
+local platform = require 'gluon.platform'
+local uci = require('simple-uci').cursor()
+
+local function get_keys(t)
+	local keys = {}
+	local n = 0
+
+	for k,_ in pairs(t) do
+		n = n+1
+		keys[n] = k
+	end
+	return keys
+end
+
+local switch_vlans = {
+	-- device identifier, lan ports, wan ports
+	["ubnt,unifiac-pro"] = {"2 0t", "3 0t"},
+}
+
+local devices = get_keys(switch_vlans)
+
+if not platform.match('ath79', 'generic', devices) then
+	return
+end
+
+local board_name = platform.get_board_name()
+local assignments = switch_vlans[board_name]
+
+uci:delete_all('network', 'switch_vlan')
+
+for vlan,ports in pairs(assignments) do
+	local wan_name = uci:add("network", "switch_vlan")
+	uci:set("network", wan_name, "device", "switch0")
+	uci:set("network", wan_name, "vlan", vlan)
+	uci:set("network", wan_name, "ports", ports)
```

Comment:
```suggestion
	uci:section("network", "switch_vlan", nil, {
		device = "switch0",
		vlan = vlan,
		ports = ports,
	}
```

Reply:
resolved in 4fbc4e0cd17fb40269dc39f0525f9cab06c0e11c